src/odemis/driver/tlfw.py

```diff
@@ -214,30 +216,30 @@ def _sendQuery(self, com):
             while True:
                 char = self._serial.read() # empty if timeout
                 if not char: # should always finish by a "> "
-                    raise IOError("Controller timeout, after receiving '%s'" % line.encode('string_escape'))
+                    raise IOError("Controller timeout, after receiving '%s'" % line.decode('utf-8'))
 
                 # normal char
                 line += char
-                if line[-2:] == "> ":
+                if line[-2:] == b"> ":
                     break
 
-            logging.debug("Received: '%s'", line.encode('string_escape'))
+            logging.debug("Received: '%s'", line.decode('utf-8'))
 
         # remove echo + suffix + new line
-        line = line[len(full_com):-2].rstrip("\r")
+        line = line[len(full_com):-2].rstrip(b"\r")
 
         # if it's an error message => raise an error
         m = re.match(self.re_err, line)
         if m:
             err = m.group(1)
             raise TLFWError("Device rejected command '%s': %s" % (com, err))
 
-        return line
+        return line.decode('utf-8')
 
     def _sendCommand(self, com):
         """
         Send a command which does not expect any answer
-        com (string): command to send (not including the ? and the \r)
```

Comment:
I decided to force the com argument to be bytes. Alternatively, we could also allow unicode and convert in this function. I thought it's a good idea to be more explicit. What do you think?

Reply:
Yes, I agree with you.
That said, I'd go even further, and would also return bytes. The function sends bytes, and returns bytes. Up the caller to do the right thing with bytes.
(In practice, that probably doesn't change much).